cmd/security-file-token-provider/testdata/docker-compose.yml

```diff
@@ -0,0 +1,129 @@
+# /*******************************************************************************
```

Comment:
Why are we embedding docker-compose files inside edgex-go for testing?  Is there an alternative that doesn't involve taking a dependency on Docker?

Reply:
Most of these security features require the cooperation of multiple different components in order to test that the feature actually does what it says.  In a lot of cases, some nonstandard configuration is needed.  In this particular case, we need to be able to inspect the state of the vault (which is normally uninspectable for security reasons) in order to tell that the code did something that we want.

Reply:
I am open to alternatives, but the two supported runtime environments are supposed to be snaps and docker.

Reply:
The other alternative is to not check these files in and include them verbatim in the PR testing instructions.

Reply:
@michaelestrin Your call as to whether I pull the scripts or leave them in.  Tingyu approved in https://github.com/edgexfoundry/edgex-go/pull/2202#pullrequestreview-332629740 but I need an approving review to merge since I had to resolve a merge conflict.

Reply:
My concern stems from the community direction to coalesce on a single referential set of docker-compose files in the https://github.com/edgexfoundry/developer-scripts/ repository.  

We've experienced difficulty in keeping the docker-compose file in https://github.com/edgexfoundry/blackbox-testing/ in sync; so much so that we've created several issues (https://github.com/edgexfoundry/developer-scripts/issues/154, https://github.com/edgexfoundry/blackbox-testing/issues/295) and are working towards eliminating it altogether.

If I understand correctly, these docker-compose files exist to facilitate integration testing of this PR (and are ultimately useful for ongoing integration testing).  My preference is to remove them from your PR (i.e. not include them in the edgex-go repository).  I would strongly encourage you to leverage them in an integration test suite -- perhaps a conversation with @frq92290 and the Q/A working group is needed to understand where and how to implement such a thing.

@bnevis-i

Reply:
@michaelestrin   Once there are users of the new functionality there would be no longer a need for the scripts.  However, this will not occur for several more weeks.   I will remove the scripts.